pyro/ops/gaussian.py

```diff
@@ -105,4 +179,6 @@ def gaussian_tensordot(x, y, dims=0):
         diff = 0.5 * nb * math.log(2 * math.pi) + 0.5 * Linvb.squeeze(-1).pow(2).sum(-1) - logdet
         log_normalizer = log_normalizer + diff
 
-    return Gaussian(log_normalizer, info_vec, precision)
+    result = Gaussian(log_normalizer, info_vec, precision)
+    assert result.dim() == x.dim() + y.dim() - dims
```

Comment:
@fehiepsi It looks like we missed a bug in the previous review, leading to a failure of this new assertion (which should be the guarantee provided by `tensordot`). Could you take a look?

Reply:
@fritzo I think that this should be `x.dim() + y.dim() - 2 * dims`. It is tested in the test.

Reply:
See [tensordot documentation](https://docs.scipy.org/doc/numpy/reference/generated/numpy.tensordot.html). The dims argument specifies how many dims of each of x and y are eliminted, i.e. the amount of overlap.